core/coreunix/metadata.go

```diff
@@ -55,3 +63,88 @@ func Metadata(n *core.IpfsNode, skey string) (*ft.Metadata, error) {
 
 	return ft.MetadataFromBytes(pbnd.Data())
 }
+
+// MetaDataMap takes a root node path and returns an unmarshalled metadata map in bytes
+func MetaDataMap(ctx context.Context, api coreiface.CoreAPI, path ipath.Path) (map[string]interface{}, error) {
+	metaData, err := MetaData(ctx, api, path)
+	if err != nil {
+		return nil, err
+	}
+
+	jmap := make(map[string]interface{})
+	err = json.Unmarshal(metaData, &jmap)
+	if err != nil {
+		return nil, err
+	}
+
+	return jmap, nil
+}
+
+// MetaData takes a root node path and returns the metadata in bytes
```

Comment:
Note, this is a helper to read the first level of the metadata dag - we intend to keep it this way.